src/diffusers/models/transformer_temporal.py

```diff
@@ -149,13 +161,28 @@ def forward(
 
         residual = hidden_states
 
-        hidden_states = hidden_states[None, :].reshape(batch_size, num_frames, channel, height, width)
-        hidden_states = hidden_states.permute(0, 2, 1, 3, 4)
+        # Apply group norm over batch_frames
+        if not self.apply_framewise_group_norm:
+            hidden_states = self.norm(hidden_states)
+            hidden_states = hidden_states[None, :].reshape(batch_size, num_frames, channel, height, width)
+            hidden_states = hidden_states.permute(0, 3, 4, 1, 2).reshape(
+                batch_size * height * width, num_frames, channel
+            )
+
+        # Apply group norm after separating batch and frames
+        else:
+            hidden_states = hidden_states[None, :].reshape(batch_size, num_frames, channel, height, width)
+            hidden_states = hidden_states.permute(0, 2, 1, 3, 4)
+
+            hidden_states = self.norm(hidden_states)
+            hidden_states = hidden_states.permute(0, 3, 4, 2, 1).reshape(
+                batch_size * height * width, num_frames, channel
+            )
 
-        hidden_states = self.norm(hidden_states)
-        hidden_states = hidden_states.permute(0, 3, 4, 2, 1).reshape(batch_size * height * width, num_frames, channel)
+        torch.save(hidden_states, "hs-notframewise.pt")
 
         hidden_states = self.proj_in(hidden_states)
+        encoder_hidden_states = encoder_hidden_states if self.use_cross_attention else None
```

Comment:
```suggestion
        encoder_hidden_states = encoder_hidden_states if self.use_cross_attention else None
```
do we need this?

Reply:
I still don't understand why we need this. If `cross_attention_dim` is None then why do we have to manually set encodre_hidden_states to `None`. This looks more like a hacky bug correction. Why do we pass `encoder_hidden_states` in the first place if we don't have cross attention?

Reply:
@patrickvonplaten 
We can omit sending the encoder hidden state to this block from the higher level blocks.  It just means the MotionModules in the UNetMotionModel cannot support cross attention at all.

Reply:
We should only add options that are needed to run the official animate diff checkpoints, all possible customizations that the user could try out should not be added (only if it becomes necessary)

Reply:
Here if the motion modules all have `temporal_cross_attention_dim` always set to None, the let's **not** give the possibility to customize it as this unnecessarily bloats the code